src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptionsExtensions.cs

```diff
@@ -58,23 +58,33 @@ public static THeaders GetHeaders<THeaders>(this OtlpExporterOptions options, Ac
         {
             // According to the specification, URL-encoded headers must be supported.
             optionHeaders = Uri.UnescapeDataString(optionHeaders);
+            ReadOnlySpan<char> headersSpan = optionHeaders.AsSpan();
 
-            Array.ForEach(
-                optionHeaders.Split(','),
-                (pair) =>
+            while (!headersSpan.IsEmpty)
+            {
+                int commaIndex = headersSpan.IndexOf(',');
+                ReadOnlySpan<char> pair;
+                if (commaIndex == -1)
                 {
-                    // Specify the maximum number of substrings to return to 2
-                    // This treats everything that follows the first `=` in the string as the value to be added for the metadata key
-                    var keyValueData = pair.Split(['='], 2);
-                    if (keyValueData.Length != 2)
-                    {
-                        throw new ArgumentException("Headers provided in an invalid format.");
-                    }
+                    pair = headersSpan;
+                    headersSpan = [];
+                }
+                else
+                {
+                    pair = headersSpan.Slice(0, commaIndex);
+                    headersSpan = headersSpan.Slice(commaIndex + 1);
+                }
 
-                    var key = keyValueData[0].Trim();
-                    var value = keyValueData[1].Trim();
-                    addHeader(headers, key, value);
-                });
+                int equalIndex = pair.IndexOf('=');
+                if (equalIndex == -1)
+                {
+                    throw new ArgumentException("Headers provided in an invalid format.");
+                }
+
+                var key = pair.Slice(0, equalIndex).ToString().Trim();
+                var value = pair.Slice(equalIndex + 1).ToString().Trim();
```

Comment:
I think that allocation can be reduced even more by utilizing https://learn.microsoft.com/en-us/dotnet/api/system.memoryextensions.trim?view=net-9.0#system-memoryextensions-trim(system-readonlymemory((system-char)))

What you should do is reverting ToString() and Trim() order.

```suggestion
                var key = pair.Slice(0, equalIndex).Trim().ToString();
                var value = pair.Slice(equalIndex + 1).Trim().ToString();
```

Reply:
Good point! Reversing the ToString() and Trim() order makes sense—it reduces allocations by working directly on ReadOnlyMemory<char>. Thanks for the insightful suggestion! I'll update the PR accordingly.